statescript/executor.go

```diff
@@ -206,13 +240,31 @@ func (l Launcher) ExecuteAll(state, action string, ignoreError bool) error {
 			}
 		}
 
-		if ret := execute(filepath.Join(dir, s.Name()), timeout); ret != 0 {
-			// In case of error scripts all should be executed.
-			if ignoreError {
-				log.Errorf("statescript: ignoring error executing '%s': %d", s.Name(), ret)
-			} else {
-				return errors.Errorf("statescript: error executing '%s': %d",
-					s.Name(), ret)
+		iet := time.Now()
+
+	retrylater:
+		for {
+			switch ret := execute(filepath.Join(dir, s.Name()), timeout); ret {
+			case 0:
+				// success. Break the loop and continue
+				break retrylater
+			case exitRetryLater:
+				if time.Since(iet) <= l.getRetryTimeout() {
+					log.Errorf("statescript: %s requested a retry", s.Name())
+					time.Sleep(l.getRetryInterval())
+					continue
+				}
+				return errors.New("statescript: retry time-limit exceeded")
```

Comment:
Because I am not sure about the overall design with using two different timeouts, I can not say for sure, but probably in this case we should check `ignoreErrors` case as well and proceed with the other scripts to execute.